swap_init.py

```diff
@@ -192,16 +198,28 @@ def resolver_setup(nameservers):
     """Writes resolv.conf file.
     """
 
-    with open("/etc/resolv.conf", 'w') as f:
-        f.write(resolver_gen(nameservers))
+    # regular resolv.conf
+    if not os.path.islink('/etc/resolv.conf'):
+        with open("/etc/resolv.conf", 'w') as f:
+            f.write(resolver_gen(nameservers, 'regular'))
+    # regular resolv.conf for resolvconf tool
     rconfdir='/etc/resolvconf/resolv.conf.d'
     if os.path.exists(rconfdir):
         with open('%s/original' % rconfdir, 'w') as f:
-                f.write(resolver_gen(nameservers))
+                f.write(resolver_gen(nameservers, 'regular'))
         rlink = '%s/tail' % rconfdir
         if os.path.exists(rlink):
             os.unlink(rlink)
         os.symlink('%s/original' % rconfdir, rlink)
+    # systemd-resolved
+    if os.path.exists('/etc/systemd/resolved.conf'):
+        dns_list=resolver_gen(nameservers, 'dnslist')
+        rpath='/etc/systemd/resolved.conf.d'
+        if not os.path.exists(rpath):
+            os.mkdir(rpath, 0755)
```

Comment:
If you want to be compatible with python 3, you can use `0o755` for octal numbers.

Reply:
@David-5-1 thanks, fixed now.